specs/AppNotifications/AppNotificationContentSpec/AppNotificationBuilder-spec.md

```diff
@@ -0,0 +1,764 @@
+
+**WinAppSDK 1.2 \<text\> Schema**:
+
+```c#
+<text lang? = string
+      placement? = "attribution"
+      hint-callScenarioCenterAlign? = boolean />
+```
+
+These attributes are abstracted with the Text component APIs:
+
+-   SetLanguage(String language)
+-   SetIsAttributionText(Boolean isAttributionText)
+-   SetIsCallScenario(Boolean isCallScenario)
+
+Below are some example usages:
+
+```c#
+AppNotificationContent()
+    .AddText(Text(L"Content")
+        .SetLanguage(L"en-US"))
+    .AddText(Text(L"Attribution Text")
+        .SetIsAttributionText(true))
+    .Show();
+```
+
+![Text Example](TextExample.png)
+
+Developers can use SetIsCallScenario(bool) to center the text like an incoming call. This feature
+will be ignored unless the AppNotification scenario is set to IncomingCall using
+AppNotificationContent::SetScenario(enum ScenarioType).
+
+```c#
+AppNotificationContent()
+    .SetScenario(ScenarioType::IncomingCall)
+    .AddText(Text(L"Content"))
+        .SetIsCallScenario(true)
+    .Show()
+```
+
+![IncomingCall Example](IncomingCallExample.png)
+
+# Button
+
+The Button component API sets up the xml for the \<action\> element. Buttons are user-clickable
+components meant to start an action from the app.
+
+**WinAppSDK 1.2 \<action\> Schema**:
+
+```c#
+<action content = string
+        arguments = string placement? = "contextMenu"
+        imageUri? = string
+        hint-inputid? = string
+        hint-buttonStyle? = "success" | "critical"
+        hint-toolTip? = string />
+```
+
+These attributes are abstracted with the Button component APIs:
+
+-   Button(String content)
+-   AddArguments(String key)
+-   AddArguments(String key, T value)
+-   SetIconUri(String iconUri)
+-   SetToolTip(String toolTip)
+-   SetIsContextMenuPlacement(Boolean isContextMenuPlacement)
+-   SetInputId(String inputId)
+
+Below are some example usages:
+
+```c#
+AppNotificationContent()
+    .AddText(Text(L"Content"))
+    .AddTextBox(TextBox(L"textBox")
+        .SetPlaceHolderContent(L"Reply"))
+    .AddButton(Button(L"Send")
+        .SetIconUri(LR"(Assets/Icons/send.png)")
+        .AddArgument(L"action", L"textReply")
+        .SetInputId(L"textBox"))
+    .Show();
+```
+
+Example result:
+![Button Example](ButtonExample.png)
+
+Developers can change the color of the button using SetButtonStyle(ButtonStyle). The two button
+styles are ButtonStyle::Success (green) and ButtonStyle::Critical (red).
+AppNotificationContent::SetUseButtonStyle(bool) must be set to true for the button to use the button
+styles.
+
+```c#
+AppNotificationContent()
+    .SetUseButtonStyle(true)
+    .AddText(Text(L"Content"))
+    .AddTextBox(TextBox(L"textBox")
+        .SetPlaceHolderContent(L"Reply"))
+    .AddButton(Button(L"Send")
+        .AddArgument(L"action", L"textReply")
+        .SetButtonStyle(ButtonStyle::Success))
+    .Show();
+```
+
+Example result:
+![Button Success Example](ButtonSuccessExample.png)
```

Comment:
Could we use this image instead, as it showcases DASH's ideal scenario for these button colors?
![image](https://user-images.githubusercontent.com/54992755/174692659-a8b19f26-5a76-4c67-9aaa-2b45521f80f0.png)

XML:
```
<toast scenario="incomingCall" useButtonStyle="true">
    <visual>
        <binding template="ToastGeneric">
            <text hint-callScenarioCenterAlign = "true">Incoming Call</text>
            <text hint-callScenarioCenterAlign = "true">Andrew Barnes</text>
            <image hint-crop="circle"  src="https://unsplash.it/100?image=883"/>
        </binding>
    </visual>
  
    <actions>
        <action
              arguments="videoCallId" 
              content="" 
              imageUri="https://www.shareicon.net/data/256x256/2015/10/17/657571_video_512x512.png" 
              hint-buttonStyle = "Success" 
              hint-toolTip = "Asnwer video call"/>
        <action 
              arguments="voiceCallId" 
              content="" 
              imageUri="https://www.nicepng.com/png/full/379-3794777_white-phone-icon-white-phone-call-icon.png" 
              hint-buttonStyle = "Success" 
              hint-toolTip = "Answer voice call"/>
        <action 
              arguments="DeclineCallId" 
              content="" 
              imageUri="https://static.thenounproject.com/png/5012-200.png" 
              hint-buttonStyle = "Critical" 
              hint-toolTip = "Decline all"/>
      </actions>
</toast>
```

Reply:
What other scenarios would these buttons be used for? Are these just for call scenarios? Does this explain why button style doesn't work when you define an inputId since it's really only meant for this scenario?